cli/tests/test_release_candidate.py

```
@@ -1970,14 +2088,163 @@ def test_local_release_harness_stages_exact_reviewed_resolvers(tmp_path: Path) -
+    )
+
+    with pytest.raises(release_candidate.CandidateError, match="lacks its completeness marker"):
+        release_candidate._validated_installer_source(name)
+
+
+def test_reviewed_installer_requires_a_defined_source(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    monkeypatch.delitem(release_candidate.INSTALLER_ASSETS, "install.sh")
+
+    with pytest.raises(release_candidate.CandidateError, match="installer source is not defined: install.sh"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)